src/assets/theme/index.tsx

```diff
@@ -1,8 +1,56 @@
 import { createMuiTheme } from '@material-ui/core';
-
+import { css, FlattenSimpleInterpolation } from 'styled-components';
 import palette from './palette';
 import typography from './typography';
 import overrides from './overrides';
+import { COLOR_MAP } from 'common/colors';
+
+interface ThemeColors {
+  green: string;
+  greenDark: string;
+  lightBlue: string;
+}
+
+const colors: ThemeColors = {
+  green: COLOR_MAP.GREEN.BASE,
+  greenDark: COLOR_MAP.GREEN.DARK,
+  lightBlue: COLOR_MAP.BLUE,
+};
+
+interface ThemeFonts {
+  subtitle: FlattenSimpleInterpolation;
+  disclaimer: FlattenSimpleInterpolation;
+  link: FlattenSimpleInterpolation;
+}
+
+/**
+ * High-level typography groups
+ */
+const fonts: ThemeFonts = {
+  subtitle: css`
```

Comment:
FWIW- I worry that "subtitle" may be too generic...  We probably have subtitles on the site that don't use this style.  That said, we can always go with this for now and then rename when we have a clear collision.

Also, it looks like we already have typography.subtitle{1,2} defined in our theme: https://github.com/covid-projections/covid-projections/blob/30a17223005d159353db9aabd180cf0b741682c9/src/assets/theme/typography.tsx#L51

Should we be updating them and using them instead?  Or is there a distinction between `fonts` and `typography` that we want to keep?

On that note, maybe just having multiple numbered subtitle entries (`subtitle1`, `subtitle2`, etc.) is an okay answer for the collision I was worried about.

Reply:
For the name, makes sense, maybe `Subtitle1` leaves us more space to have variants in the future. Using fonts instead of the Typography directly allows us to do two things

- Have a better API to export and reuse styles
- Better communicate (in the code) what is the set of typography blocks we want to use

### Better API to export styles

The theme variables are a bit too low-level, and they don't directly provide a way to apply all of them at the same time with the flexibility that we would need. For the Markdown component, for example, we need to define the styling of the content by element type, it would be ideal to just import those styles instead of re-typing them each time:

```tsx
import ReactMarkdown from 'react-markdown';
import theme from 'assets/theme';

export const MarkdownBlock = styled(ReactMarkdown)`
  {/* using an exported css function */}
  h1 {
    ${themes.fonts.heading1};
  }

  {/* using theme variables */}
  h2 {
    font-size: ${themes.typography.h2.fontSize};
    font-family: ${themes.typography.h2.fontFamily};
    {/* etc... */}
  }
`
```

The Heading2 and Link component, for example, is used both inside markdown and outside, and having css functions to grab the styles directly help us centralize and reuse them in both contexts easily.

### Better communicate our typography building blocks

Having the CSS functions being exported as part of the theme allows us to still use the Typography component (to keep the good things, like the ability to set alignment using props).

```tsx
export const Heading1 = styled(Typography)
   .attrs(props => ({ component: 'h1' }))`
      ${theme.fonts.heading1};
   `;
```

which is more explicit (and less verbose) in the client code than using Typography directly.

```tsx
<Heading1 align="center">Title</Heading1>

{/* something like can be confusing */}
<Typography align="center" component="h1" variant="h2">
Title
</Typography>
```

Ideally, we want to have a small-ish set of font styles to be used across the app, and using this structure will help us be more thoughtful when adding a new variant. It would be nice to have components for: Heading1, Heading2, Heading3, Subtitle1, Paragraph, ChartCaption, ChartDisclaimer, Blockquote, Link among others. 

Neither of these arguments is super strong against using Typography or typography variables from the theme directly, it just makes is a bit more convenient (in my opinion). I'm open to other solutions as well, let me know if this makes sense!

Reply:
As a side note, subtitles are a bit tricky, because semantically, they belong to the header, but they are styled differently and not always present. 

Using headings to implement them is not correct (Material UI uses H2) and can break the hierarchy of the document, I think that using a block element is probably fine, but I haven't found best practices for them.

Reply:
Quick question on the typography example:
```html
{/* something like can be confusing */}
<Typography align="center" component="h1" variant="h2">
```

Why would variant h2 be remapped to h1 there? I guess I would expect that to just be:
```html
<Typography variant="h1" align="center">
```

Reply:
We could, but defining that at the theme level would apply that to all the h1 elements in the app.

Reply:
I think that makes sense! 

Did we explore generating typography compliant code from React Markdown? There seems to be an [example of generating typography elements](https://github.com/mui-org/material-ui/blob/master/docs/src/pages/getting-started/templates/blog/Markdown.js) from markdown (albeit using a slightly different package).  The [thread](https://github.com/mui-org/material-ui/issues/12290) goes into a bit more detail (with a couple alternatives).

My overall thinking here is that it seems like it's a bit more confusing to have to maintain both ways of defining an `h1` (and other similar elements).  But i'm just getting a better sense of the implications of either choice, lmk what I'm missing :)

Reply:
It sorta' makes sense (it would probably make a lot more sense if I actually understood material-ui theming 😁 )!  I guess `theme.typography` is a builtin material-ui feature that can be used via the builtin `<Typography>` component?  Whereas `theme.fonts` is our own invention that lets us be more explicit and direct.

That seems fine to me, I think...  Though perhaps we should at least add a comment to make it clearer what parts of `theme` are "built-in" material-ui variables and what parts are just our own invention.  We could also use a naming convention...  like `theme.can.*` (e.g. `theme.can.fonts` and `theme.can.colors`) would be our own inventions.

Beyond making that distinction clearer, I do wonder if it makes sense to try to still leverage `typography` and other low-level builtin material-ui theming concepts where it makes sense.  Like we move as much of the subtitle properties as we can directly into `theme.typography.subtitle1` and then reference that from our definition for `theme.fonts.subtitle1` so we don't end up with two unrelated and divergent "definitions" for what our subtitles should look like.  But again, I only sort of understand what's going on, so not 100% sure if this makes sense.